### ISSUE TITLE
Automate "What's New" popup content

### ISSUE THREAD

**Issue by cewert** [jellyfin-archive/jellyfin-roku-legacy#1103]
## Is your feature request related to a problem? Please describe
Every time we push a release, we need to manually generate and review a json changelog file. This slows down releases and is a burden for maintainers

## Describe the solution you'd like
Create a github workflow to automatically generate the .json file when unstable is merged with master for a release. Any tweaks or manual review would be done in the PR. If we are unable to automatically generate this file I think we should use the solution below as a compromise.

## Describe alternatives you've considered
Change the popup to only show the latest version (i.e. Welcome to v1.6.4!) and encourage users to visit github/reddit/our blog for more details. This would still keep users informed about what version they are running and also completely eliminate the manual process of creating/reviewing/approving the .json file. The version number could either be updated during the version bump for release or automatically using a workflow when unstable is merged with master

## Additional context
First discussed in #916

**Comment by 1hitsong** [comment]
I'm not a fan of the alternative. I don't consider generating the Whats New content very cumbersome. In fact, it's mostly adjusting and reworking items that are automatically pulled in from a script that pulls merged PRs from Github. It takes maybe 10 minutes at the most.

I don't mind continuing to do this process for each release. If I get hit by a bus and you don't want to maintain it, then y'all can have a discussion about removing it 😆

**Comment by cewert** [comment]
> I don't mind continuing to do this process for each release.

You told Jimi in the [original PR](https://github.com/jellyfin/jellyfin-roku/pull/916#issuecomment-1367379845) before he approved it that the plan was to automate this file. Is that no longer the case? You just want to generate this file manually forever now?

> It takes maybe 10 minutes at the most.

Maybe it only took you 10min to generate the file but it had stuff included that didn't need to be - it had errors. It still needed manually reviewed and edited which takes time. If you're willing to waste your time doing it go for it but it puts the team in a position where we can't push a release without you. Which you have to admit isn't ideal. Plus as I mentioned before, it's not just your time since the file needs reviewed and approved.

I noticed you just merged the what's new content PR #1090 but I don't understand why? There's going to be at least one more bugfix merged in before release. So now we're going to have to edit that file again before release, right?

**Comment by 1hitsong** [comment]
>You told Jimi in the https://github.com/jellyfin/jellyfin-roku/pull/916#issuecomment-1367379845 before he approved it that the plan was to automate this file. Is that no longer the case? You just want to generate this file manually forever now?

Nope. I meant when/if automation occurs, I don't mind doing the manual work.

> If you're willing to waste your time doing it go for it but it puts the team in a position where we can't push a release without you. Which you have to admit isn't ideal. Plus as I mentioned before, it's not just your time since the file needs reviewed and approved.

It's just like all the other PRs. It'll take a little time, but not very much.

> I noticed you just merged the what's new content PR https://github.com/jellyfin/jellyfin-roku/pull/1090 but I don't understand why? There's going to be at least one more bugfix merged in before release. So now we're going to have to edit that file again before release, right?

Which bug fix? I thought we were feature locked?

**Comment by cewert** [comment]
> Nope. I meant when/if automation occurs, I don't mind doing the manual work.

You said this:
> My thought was we'd start with it manual until one of us made it automatic - if it's possible.

Did you look into if it's possible at all? Do you plan on looking into it? Why did you change your mind about automating this file? The whole reason I made this issue was to track your plan to automate this file.

> Which bug fix? I thought we were feature locked?

We are feature locked. This is a bug fix. The whole reason we waited to push a release was to try and find more bugs - which you were the one to suggest doing actually. I'm not sure where the confusion is coming from?

The bug is the one jimi found in matrix chat yesterday that we spent half a day troubleshooting. There's an [open PR for it](https://github.com/jellyfin/jellyfin-roku/pull/1101) with an [issue attached](https://github.com/jellyfin/jellyfin-roku/issues/1099). I even put it in the [project I made for the next release](https://github.com/jellyfin/jellyfin-roku/projects/5) that I told you about in our roku team chat weeks ago.

**Comment by 1hitsong** [comment]
> Did you look into if it's possible at all? Do you plan on looking into it? Why did you change your mind about automating this file? The whole reason I made this issue was to track your plan to automate this file.

Nope. Haven't looked at if it's possible. Since doing it manually isn't a bit deal, I put automation on the back burner to work on what I see as more important items, such as getting MVP playlist support. I haven't changed my mind about automating it, my comment is specifically on the alternative being to remove it if we can't automate it.

If we can't automate it, then the alternative is I'll keep doing it manually. That's all I meant.

**Comment by cewert** [comment]
> Nope. Haven't looked at if it's possible. Since doing it manually isn't a bit deal, I put automation on the back burner to work on what I see as more important items, such as getting MVP playlist support. I haven't changed my mind about automating it, my comment is specifically on the alternative being to remove it if we can't automate it.

Understood, thanks for explaining.

> If we can't automate it, then the alternative is I'll keep doing it manually. That's all I meant.

Agree to disagree on this one. That is the worst out of all ~~3~~ 4 options IMO and does nothing to address the problem stated in my issue: `Every time we push a release, we need to manually generate and review a json changelog file. This slows down releases and is a burden for maintainers`

1. Alternative solution - completely eliminates the problem while still providing value to users by alerting them to new releases. I would even argue that this slimmer version shouldn't have a user setting and should be shown to all users because why not? No need to scroll just hit any key and it goes away but still told me the app has been updated. Cool. 
2. Revert the PR - completely eliminates the problem but also completely eliminates the benefits of having the popup
3. Proposed solution - slightly eliminates the problem and eliminates the team's reliance on you to run the report (assuming your the only one who wants to do it). 
4. You doing it manually - does nothing to address the problem


I also want to point out that no other roku app I use does anything like this. Some show the version in settings or menus etc. but i can't think of ANY other roku app that I use that has a popup just to alert users about every new version. Can you? I'm curious how other apps deal with this.

Edit: added the 4th option - reverting the PR
